slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -101,18 +101,25 @@ ::grpc::Status SLAMServiceImpl::GetPointCloudMap(
     try {
         std::shared_lock optimization_lock{optimization_shared_mutex,
                                            std::defer_lock};
-        if (optimization_lock.try_lock()) {
+        if ((optimization_lock.try_lock()) && (!localization_map_ready)) {
             // We are able to lock the optimization_shared_mutex, which means
             // that the optimization is not ongoing and we can grab the newest
             // map
             GetLatestSampledPointCloudMapString(pointcloud_map);
             std::lock_guard<std::mutex> lk(viam_response_mutex);
             latest_pointcloud_map = pointcloud_map;
         } else {
-            // We couldn't lock the mutex which means the optimization process
-            // locked it and we need to use the backed up latest map
-            LOG(INFO)
-                << "Optimization is occuring, using cached pointcloud map";
+            // Either we are in localization mode or we couldn't lock the mutex
+            // which means the optimization process locked it and we need to use
+            // the backed up latest map
+            if (localization_map_ready) {
```

Comment:
[minor]
```suggestion
            if (action_mode == ActionMode::LOCALIZING) {
```

Reply:
I agree with Kat's suggestion. To me, whether cartogropher is in localization mode should only be determined  exclusively by the action mode.

Reply:
was already making changes so I added this to my commit